src/HttpMock/RequestHandler.cs

```diff
@@ -121,5 +121,9 @@ public ReceivedRequest LastRequest()
 		{
 			return _observedRequests.Peek();
 		}
+
+		public ReceivedRequest GetRequestAt(int index){
+			return _observedRequests.ElementAt(index);

```

Comment:
Suggest returning the collection behind the _observedRequests queue. It will provided the flexibility to more complex matching in the tests, instead of having to loop over.

So instead of the GetRequestAt , have a GetObservedRequests()